src/types/redis_hyperloglog.cc

```diff
@@ -28,6 +28,25 @@
 
 namespace redis {
 
+namespace {
+template <typename ElementType>
+std::vector<nonstd::span<const uint8_t>> TransformToSpan(const std::vector<ElementType> &registers) {
+  std::vector<nonstd::span<const uint8_t>> register_segments;
+  register_segments.reserve(kHyperLogLogSegmentCount);
+  for (const auto &register_segment : registers) {
+    if (register_segment.empty()) {
+      // Empty segment
+      register_segments.emplace_back();
+      continue;
+    }
+    // NOLINTNEXTLINE
+    const uint8_t *segment_data_ptr = reinterpret_cast<const uint8_t *>(register_segment.data());
```

Comment:
should we consider different endian here?

Reply:
We always uses lsb here

Reply:
then why we need reinterpret_cast here?

Reply:
This means casting `const char*` to `const uint8_t*`

Reply:
ahh got it